pygmsh/built_in/geometry.py

```diff
@@ -192,6 +192,29 @@ def add_physical_volume(self, volumes, label=None):
         self._add_physical('Volume', volumes, label=label)
         return
 
+    def add_transfinite_lines(self, lines, size):
+        self._GMSH_CODE.append(
+            'Transfinite Line {{{0}}} = {1};'.format(', '.join([l.id for l in lines]), size
+            ))
+
+    def add_transfinite_surface(self, surface, size=None):
+        # a transfinite surface can only have 4 sides
+        assert surface.num_edges == 4
+        if size is not None:
+            if hasattr(surface, 'line_loop'):
```

Comment:
I don't know how to check for instances of `Polygon` since that class is defined only inside the `add_polygon()` function. do you know a way to do it? should I promote `Polygon` to be a class outside that function?

Reply:
![dewit](https://user-images.githubusercontent.com/181628/38510402-41eca20c-3c25-11e8-9549-36a65538e6ba.gif)